app/models/order_updater.rb

```diff
@@ -84,11 +84,9 @@ def failed_payments?
   # shipments using a shipping method that doesn't require address, such us
   # a pickup.
   def shipping_address_from_distributor
-    shipments.each do |shipment|
-      shipping_method = shipment.shipping_method
-      next if shipping_method.require_ship_address
+    return if order.shipment.andand.shipping_method.blank?
+    return if order.shipment.shipping_method.require_ship_address
```

Comment:
I was just discussing the use of andand in another PR with @mkllnk 
maybe:
```ruby
    return if order.shipment.blank?
    return if order.shipment.shipping_method.blank?
    return if order.shipment.shipping_method.require_ship_address
```
or simply?
```ruby
    return if order.shipment.andand.shipping_method.andand.require_ship_address
```

Reply:
Or we implement the law of Demeter and then write:
```rb
return if order.require_ship_address?
```

:smile: